pyserini/dsearch/_dsearcher.py

```diff
@@ -192,6 +192,36 @@ def encode(self, query: str):
             return super().encode(query)
 
 
+class AutoQueryEncoder(QueryEncoder):
```

Comment:
Is the intention here that AutoQueryEncoder will replace other model specific ones?

Reply:
yes, the most common dense encoder is taking CLS from the encoder model
We can try to group these models as Auto I think?

Reply:
File an issue, doesn't need to be part of this PR.

Reply:
> File an issue, doesn't need to be part of this PR.

Should I rename it to something like `DistilBertQueryEncoder` in this PR?

Reply:
Nah... I think it's okay for master to be in a slightly inconsistent state as long as we fix if before the next release.